test/test_optim.py

```diff
@@ -357,6 +357,10 @@ def test_sparse_adam(self):
         )
         with self.assertRaisesRegex(ValueError, "Invalid beta parameter at index 0: 1.0"):
             optim.SparseAdam(None, lr=1e-2, betas=(1.0, 0.0))
+        with self.assertRaisesRegex(ValueError, "SparseAdam requires dense parameter tensors"):
+            optim.SparseAdam([torch.zeros(3, layout=torch.sparse_coo)])
+        with self.assertRaisesRegex(ValueError, "SparseAdam requires dense parameter tensors"):
+            optim.SparseAdam([{"params": torch.zeros(3, layout=torch.sparse_coo)}])
```

Comment:
"params" should be a list, so the correct usage is:
`optim.SparseAdam([{"params": [torch.zeros(3, layout=torch.sparse_coo)]}])`

The documentation for that is here: https://pytorch.org/docs/master/optim.html#per-parameter-options